File: wlmedia_sample/wlmedia/src/main/java/com/ywl5320/wlmedia/enums/WlScaleType.java
package com.ywl5320.wlmedia.enums;

/**
 * author : ywl5320
 * e-mail : ywl5320@163.com
 * desc   : wlmedia
 * date   : 2024/9/5
 */
public enum WlScaleType {

    WL_SCALE_16_9("WL_SCALE_16_9", 16, 9, "scale video width and height use 16:9"),
    WL_SCALE_4_3("WL_SCALE_4_3", 4, 3, "scale video width and height use 4:3"),
    WL_SCALE_FIT("WL_SCALE_FIT", 0, 0, "scale video width and height fit"),
    WL_SCALE_MATCH("WL_SCALE_MATCH", -1, -1, "scale video width and height match"),
    WL_SCALE_FILL("WL_SCALE_MATCH", -2, -2, "scale video width and height fill");

    private String key;
    private int scaleWidth = 0;
    private int scaleHeight = 0;
    private String desc;

    WlScaleType(String key, int scaleWidth, int scaleHeight, String desc) {
        this.key = key;
        this.scaleWidth = scaleWidth;
        this.scaleHeight = scaleHeight;
        this.desc = desc;
    }

    public String getKey() {
        return key;
    }

    public int getScaleWidth() {
        return scaleWidth;
    }

    public int getScaleHeight() {
        return scaleHeight;
    }

    public String getDesc() {
        return desc;
    }
}

File: wlmedia_hm/libwlmedia/src/main/ets/com/ywl5320/wlmedia/enums/WlScaleType.ts
/**
 * author : ywl5320
 * e-mail : ywl5320@163.com
 * desc   : wlmedia
 * date   : 2025/1/6
 */
export enum WlScaleType {
  WL_SCALE_16_9 = "16:9",
  WL_SCALE_4_3 = "4:3",
  WL_SCALE_FIT = "FIT(长宽适配)",
  WL_SCALE_MATCH = "MATCH（拉伸填充）"
}

interface WlScaleValue {
  scaleWidth: number;
  scaleHeight: number;
  msg: string
}

export const WlScaleTypeInfoMap: Record<WlScaleType, WlScaleValue> = {
  [WlScaleType.WL_SCALE_16_9]: { scaleWidth: 16, scaleHeight: 9, msg: "16:9" },
  [WlScaleType.WL_SCALE_4_3]: { scaleWidth: 4, scaleHeight: 3, msg: "4:3" },
  [WlScaleType.WL_SCALE_FIT]: { scaleWidth: 0, scaleHeight: 0, msg: "FIT(长宽适配)" },
  [WlScaleType.WL_SCALE_MATCH]: { scaleWidth: -1, scaleHeight: -1, msg: "MATCH（拉伸填充）" }
};
